apps/comps/e2e/tests/perps-competition.test.ts

```diff
@@ -4630,4 +4630,318 @@ describe("Perps Competition", () => {
+      agentIds: [agent.id],
+      perpsProvider: {
+        provider: "hyperliquid",
+        apiUrl: "http://localhost:4568",
+      },
+    });
+
+    expect(competition2.success).toBe(true);
+    const comp2Id = competition2.competition.id;
+
+    // Sync positions for competition 2
+    // Before the fix, this would FAIL with unique constraint violation
+    // because providerPositionId was globally unique
+    await perpsDataProcessor.processPerpsCompetition(comp2Id);
+    await wait(500);
+
+    // Verify position was stored for competition 2 via API
+    const comp2Positions = await agentClient.getPerpsPositions(comp2Id);
+    expect(comp2Positions.success).toBe(true);
+    const typedComp2Positions = comp2Positions as PerpsPositionsResponse;
+    expect(typedComp2Positions.positions).toHaveLength(1);
+
+    const btcPositionComp2 = typedComp2Positions.positions[0];
+    expect(btcPositionComp2?.marketSymbol).toBe("BTC");
+    expect(btcPositionComp2?.isLong).toBe(true);
+
+    // Both competitions should have the same position data (same wallet, same asset)
+    expect(btcPositionComp1?.marketSymbol).toBe(btcPositionComp2?.marketSymbol);
+    expect(btcPositionComp1?.isLong).toBe(btcPositionComp2?.isLong);
+
+    // Clean up
+    await adminClient.endCompetition(comp2Id);
+  });
+
+  test("should allow same position in multiple sequential Symphony competitions", async () => {
+    // Same test as above but with Symphony provider to ensure fix works for both
+
+    const adminClient = createTestClient(getBaseUrl());
+    await adminClient.loginAsAdmin(adminApiKey);
+
+    // Register agent with wallet that has BTC/ETH positions configured in Symphony mock
+    const { agent, client: agentClient } =
+      await registerUserAndAgentAndGetClient({
+        adminApiKey,
+        agentName: "Symphony Multi-Competition Agent",
+        agentWalletAddress: "0x1111111111111111111111111111111111111111", // BTC/ETH positions
+      });
+
+    // ===== COMPETITION 1 =====
+    const competition1 = await startPerpsTestCompetition({
+      adminClient,
+      name: `Symphony Position Constraint Test Comp 1 - ${Date.now()}`,
+      agentIds: [agent.id],
+      perpsProvider: {
+        provider: "symphony",
+        initialCapital: 1000,
+        apiUrl: "http://localhost:4567",
+      },
+    });
+
+    expect(competition1.success).toBe(true);
+    const comp1Id = competition1.competition.id;
+
+    // Sync positions
+    await perpsDataProcessor.processPerpsCompetition(comp1Id);
+    await wait(500);
+
+    // Verify positions stored via API
+    const comp1Positions = await agentClient.getPerpsPositions(comp1Id);
+    expect(comp1Positions.success).toBe(true);
+    const typedComp1Positions = comp1Positions as PerpsPositionsResponse;
+    expect(typedComp1Positions.positions).toHaveLength(2); // BTC and ETH
+
+    // End competition 1
+    await adminClient.endCompetition(comp1Id);
+    await wait(1000);
```

Comment:
I wonder if we need to use any of these `wait` methods? 

I know we had to do this back in the day, but maybe it's not needed anymore?